### ISSUE TITLE
Plan nudge on domain purchase doesn't show for all users

### ISSUE THREAD

**Issue by carinapilar** [Automattic/wp-calypso#56242]
### What

When a user adds a domain to the checkout, on a site with a free plan, we could add a nudge there to buy a plan.

### Why

This is what it currently shows on the checkout:

![AddGetAPlanNudge](https://user-images.githubusercontent.com/16104522/133307341-0bd4704d-cf44-4413-8a57-744c473122c9.png)

It's common to get users confused with the fact that can't use a domain as primary after getting one on a free site. It feels that we could add a nudge on this purchase, explaining primary addresses and the features from paid plans.

**Comment by DavidRothstein** [comment]
This already exists -- here's what I see:

![Screenshot from 2021-09-24 11-11-33](https://user-images.githubusercontent.com/235183/134699704-11199a88-ba54-46cc-afe8-396277b49a37.png)

Are there steps to reproduce that cause that not to appear?  I tried it a couple of times and so far it always showed for me.

**Comment by carinapilar** [comment]
> Are there steps to reproduce that cause that not to appear?

Hm, interesting! I tested with different accounts now and also see that message. I first thought it could be the currency, but I tested with USD and BRL and the message was still there. I would guess it might be related to my A8c account (which I used first) being PWPO exempt: P6Ya1B-Nh-p2.

**Comment by DavidRothstein** [comment]
>I would guess it might be related to my A8c account (which I used first) being PWPO exempt

Looks like you're right!  Here's the code that suppresses the plan nudge for those users: https://github.com/Automattic/wp-calypso/blob/b11f2d49590d814c0b53efadaeb405d9e850b8bd/client/my-sites/checkout/cart/cart-free-user-plan-upsell.jsx#L177

It was originally added here: https://github.com/Automattic/wp-calypso/pull/39610

I am not sure if there was a reason (I wonder if it was a temporary measure when the nudge was being tested, but was never removed).

One thing we discussed in our planning call was that it seems like the current wording of the nudge works just as well for PWPO exempt users as regular users, but one could also imagine rewriting the copy a bit (to emphasize some of the specific features of a Personal Plan, including being able to set the domain as primary) and that wording might not make sense for a user who is PWPO exampt.

We added this to our backlog, but it's not a high priority issue for us to investigate further right now. I'm also cc-ing @Automattic/nomado.
